spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -155,6 +155,7 @@ public void executeTask(String taskName, Map<String, String> taskDeploymentPrope
 					+ taskName);
 		}
 		taskExecutionRepository.updateExternalExecutionId(taskExecution.getExecutionId(), id);
+		return taskExecution.getExecutionId();
```

Comment:
Given we already return the external executionId (Deployment ID) when launching and now returning the `executionId`, do these all mean the client has to wait for the launch request to complete at least in case of `synchronous` task launching?

Also, based on the `javadoc` for the TaskLauncher's `launch`:

> Launch a task for the provided AppDeploymentRequest. The returned id may later be used with cancel(String) or status(String) to cancel a task or get its status, respectively. Implementations may perform this operation asynchronously; therefore a successful launch may not be assumed upon return. To determine the status of a launch, invoke status(String).

I believe `launching` of the task always needs to be `asynchrounous` in order for the task to be **canceled**